From 7b8a7371c1517750a481414ebd886cb9d1f02405 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:39:29 -0400
Subject: [virt] kvm: Remove SMAP bit from CR4_RESERVED_BITS

Message-id: <1406040016-3289-175-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86253
O-Subject: [RHEL7 PATCH v2 174/221] KVM: Remove SMAP bit from CR4_RESERVED_BITS
Bugzilla: 1084616 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Feng Wu <feng.wu@intel.com>

This patch removes SMAP bit from CR4_RESERVED_BITS.

Signed-off-by: Feng Wu <feng.wu@intel.com>
Signed-off-by: Marcelo Tosatti <mtosatti@redhat.com>
(cherry picked from commit 56d6efc2de5fcf76d3c7b33a7671bc04c53cb0e5)
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/include/asm/kvm_host.h b/arch/x86/include/asm/kvm_host.h
index fcaf9c9..7de069a 100644
--- a/arch/x86/include/asm/kvm_host.h
+++ b/arch/x86/include/asm/kvm_host.h
@@ -60,7 +60,7 @@
      | X86_CR4_PSE | X86_CR4_PAE | X86_CR4_MCE     \
      | X86_CR4_PGE | X86_CR4_PCE | X86_CR4_OSFXSR | X86_CR4_PCIDE \
      | X86_CR4_OSXSAVE | X86_CR4_SMEP | X86_CR4_FSGSBASE \
-     | X86_CR4_OSXMMEXCPT | X86_CR4_VMXE))
+     | X86_CR4_OSXMMEXCPT | X86_CR4_VMXE | X86_CR4_SMAP))
 
 #define CR8_RESERVED_BITS (~(unsigned long)X86_CR8_TPR)
 
-- 
1.7.1